backend/app/routers/session.py

```diff
@@ -46,8 +48,21 @@ async def delete_session(
     await session_service.delete_session(db_session, session_id, current_user.id)
     return session_id
 
+
 @router.get("/{session_id}/ranking", response_model=list[UserSessionRankingOut])
 async def get_session_ranking(
     db_session: Annotated[AsyncSession, Depends(get_session)], session_id: str
 ) -> list[UserSessionRankingOut]:
     return await session_service.get_session_ranking(db_session, session_id)
+
+
+@router.post("/{session_id}/ranking/{user_name}/submit", response_model=QuizCheckOut)
+async def check_solution(
+    db_session: Annotated[AsyncSession, Depends(get_session)],
+    session_id: str,
+    user_name: str,
+    quiz_id: str,
+    files_in: QuizCheckIn,
+) -> QuizCheckOut:
+    await session_service.asign_solution_to_user_session(db_session, session_id, user_name)
+    return await quiz_check_service.check_files(db_session, quiz_id, files_in)
```

Comment:
AFAIR Client didn't want to show the results to the student after submission. I'm wondering if we should limit the response 🤔 What do you think?

Reply:
That's true, however we would need to create some mechanism for the admin, to allow students to see their score, so IMO for the time being we can stay with instant scoring.

Reply:
Right. So let's leave it as it is 😄